### PR TITLE
Court model

### PR DESCRIPTION
**Pull request by actlikewill**
- [x] Create court model
- [x] Migrate author data to court Model
- [x] Add court field to elastic search index
- [x] Add court facet to search query
- [x] Remove author model from judgment model
- [x] Add court to index

Ref #428

### CODE REVIEW

**Comment by actlikewill** [comment]
@nickmwangemi @longhotsummer Run migrations and check for errors. Note migration operations are irreversible! 😲

File: peachjam/migrations/0024_court_model.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+import django.db.models.deletion
+from django.db import migrations, models
+
+
+def forwards(apps, _):
+    judgment_model = apps.get_model("peachjam", "Judgment")
+    new_court_class = apps.get_model("peachjam", "CourtClass")
+    new_court = apps.get_model("peachjam", "Court")
+
+    if apps.is_installed("liiweb"):
```
Comment (longhotsummer) [inline peachjam/migrations/0024_court_model.py:12]: nice

File: peachjam/views/home.py [hunk 2]
```diff
@@ -24,15 +25,16 @@ def get(self, request, *args, **kwargs):
 
         authors = Author.objects.exclude(
             Q(genericdocument__isnull=True),
-            Q(judgment__isnull=True),
             Q(legalinstrument__isnull=True),
         )
 
+        courts = Court.objects.exclude(Q(judgment__isnull=True))
```
Comment (longhotsummer) [inline peachjam/views/home.py:31]: I think this is being too smart, just include all the courts configured in the system.

**Comment by longhotsummer** [comment]
The lawlibrary court detail page fails to load http://localhost:8000/judgments/ZASC/

Cannot resolve keyword 'author' into field. Choices are: additional_citations, case_name, case_numbers, citation, citation_links, content_html, content_html_is_akn, coredocument_ptr, coredocument_ptr_id, court, court_id, created_at, date, doc_type, expression_frbr_uri, flynote, frbr_uri_actor, frbr_uri_date, frbr_uri_doctype, frbr_uri_number, frbr_uri_subtype, headnote_holding, hearing_date, id, images, judges, jurisdiction, jurisdiction_id, language, language_id, locality, locality_id, media_summaries, mnc, serial_number, source_file, source_url, taxonomies, title, toc_json, updated_at, work, work_frbr_uri, work_id

**Comment by longhotsummer** [comment]
Migration worked well for me 👍 

Please also add Court as a filter and a column for judgments in the admin view.

File: peachjam/forms.py [hunk 3]
```diff
@@ -107,8 +108,11 @@ def filter_queryset(self, queryset, exclude=None):
         if alphabet and exclude != "alphabet":
             queryset = queryset.filter(title__istartswith=alphabet)
 
-        if author and exclude != "author":
-            queryset = queryset.filter(author__name__in=author)
+        if authors and exclude != "author":
+            queryset = queryset.filter(author__name__in=authors)
+
+        if courts and exclude != "courts":
+            queryset = queryset.filter(court__name__in=courts)
```
Comment (actlikewill) [inline peachjam/forms.py:116]: @longhotsummer Not sure if this code is working as expected. Should it narrow out the facet list after filtering on a parameter or not?

Comment (actlikewill) [reply peachjam/forms.py:116]: > Migration worked well for me 👍
> 
> Please also add Court as a filter and a column for judgments in the admin view.

Missed this 😢 Let me add now.

Comment (actlikewill) [reply peachjam/forms.py:116]: > 

Done

Comment (longhotsummer) [reply peachjam/forms.py:116]: > @longhotsummer Not sure if this code is working as expected. Should it narrow out the facet list after filtering on a parameter or not?

You've got it doing the correct thing. It applies the courts filter unless explicitly told to exclude it.

**Comment by actlikewill** [comment]
@longhotsummer Are we ok to merge this?

**Comment by longhotsummer** [comment]
> @longhotsummer Are we ok to merge this?

Yes, please do!

**Pull request merged**